src/compiler/ruby_generator_string-inl.h

```diff
@@ -125,6 +125,31 @@ inline grpc::string RubyTypeOf(const grpc::string& a_type,
   }
 }
 
+// RubyPackage gets the ruby package in either proto or ruby_package format
+inline grpc::string RubyPackage(const grpc::protobuf::FileDescriptor* file) {
+  grpc::string package_name = file->package();
+  if (file->options().has_ruby_package()) {
+    package_name = file->options().ruby_package();
+
+    // If :: is in the package convert the Ruby formated name
+    //    -> A::B::C
+    // to use the dot seperator notation
+    //    -> A.B.C
+    package_name = ReplaceAll(package_name, "::", ".");
+  }
+  return package_name;
+}
+
+// RubyTypeOf gets the fully qualified Ruby name of the given descriptor
+inline grpc::string RubyTypeOf(const grpc::protobuf::Descriptor* descriptor,
```

Comment:
nit: can we please remove the other (currently overloaded) `RubyTypeOf` definition - it looks like it's now dead code.

Reply:
This new `RubyTypeOf` uses the existing `RubyTypeOf` function. I could rename it but I reused the same since they both do the same thing with different inputs.

Reply:
Sorry, my bad - realize that's not dead code. However, I'd prefer to avoid overloading `RubyTypeOf`.

Can we instead just merge this code into the existing `RubyTypeOf`? E.g., we could change the existing `RubyTypeOf` to take these params here, and change the top to something like:

```
std::string proto_type = descriptor->full_name();
if (descriptor->file()->options().has_ruby_package()) {
   proto_type = RubyPackage(descriptor->file()) + "." + descriptor->name();
}
grpc::string res(proto_type);
....
```

Reply:
Sure. It looks like the original method isn't used anywhere else so merging makes sense.